exporter/lokiexporter/next_exporter_test.go

```diff
@@ -62,8 +62,8 @@ func TestPushLogData(t *testing.T) {
 			hints: map[string]interface{}{
 				"loki.resource.labels": "host.name",
 			},
-			expectedLabel: `{exporter="OTLP", host.name="guarana"}`,
-			expectedLine:  `{"traceid":"01020304000000000000000000000000","resources":{"region.az":"eu-west-1a"}}`,
+			expectedLabel: `{exporter="OTLP"}`,
```

Comment:
This looks wrong: the hint states that `host.name` should be added to the label, but looks like it's now not part of the label set, being part of the body only? This makes me wonder if we had a problem after all...

Reply:
Looking closer at the code, I believe this is more of a documentation issue. Each entry should be able to decide which resource attributes it wants to promote to a label. We _could_ improve this later so that the hint could be looked up under the resource attributes as well, but we should not remove the ability for records to select labels.

The example should then be fixed, to look like this:

```yaml
processors:
  attributes:
    actions:
    - action: insert
      key: loki.resource.labels
      value: host_name, pod_name
    - action: insert
      key: loki.attribute.labels
      value: http_status_code

    # the following attributes are added manually here in the example, but would
    # probably be added by other processors or straight from the source
    - action: insert
      key: http_status_code
      value: 500

  resource:
    attributes:
    # the following attributes are added manually here in the example, but would
    # probably be added by other processors or straight from the source
    - action: insert
      key: host_name
      value: guarana
    - action: insert
      key: pod_name
      value: guarana-pod-01
```

Reply:
so that means, the following configuration is invalid? use cannot promote to label directly from `resource.attributes` ?
```
     resource:
        attributes:
        - action: insert
          key: loki.resource.labels
          value: cluster_name
```

Reply:
Right now, it won't work. We can change the processor so that this works, in addition to adding the hint as a record attribute.

Reply:
are you talking something like following:
```go
func convertAttributesAndMerge(logAttrs pcommon.Map, resAttrs pcommon.Map) model.LabelSet {
	out := defaultExporterLabels
        // add this?
	if resourcesToLabel, found := resAttrs.Get(hintResources); found {
		labels := convertAttributesToLabels(resAttrs, resourcesToLabel)
		out = out.Merge(labels)
	}

	// get the hint from the log attributes, not from the resource
	// the value can be a single resource name to use as label
	// or a slice of string values
	if resourcesToLabel, found := logAttrs.Get(hintResources); found {
		labels := convertAttributesToLabels(resAttrs, resourcesToLabel)
		out = out.Merge(labels)
	}

	if attributesToLabel, found := logAttrs.Get(hintAttributes); found {
		labels := convertAttributesToLabels(logAttrs, attributesToLabel)
		out = out.Merge(labels)
	}

	// get tenant hint from resource attributes, fallback to record attributes
	// if it is not found
	if resourcesToLabel, found := resAttrs.Get(hintTenant); !found {
		if attributesToLabel, found := logAttrs.Get(hintTenant); found {
			labels := convertAttributesToLabels(logAttrs, attributesToLabel)
			out = out.Merge(labels)
		}
	} else {
		labels := convertAttributesToLabels(resAttrs, resourcesToLabel)
		out = out.Merge(labels)
	}

	return out
}
```

Reply:
Yes, something along those lines.